src/config/index.js

```diff
@@ -1,92 +1,518 @@
 const fs = require("fs");
-const Path = require("path");
 const YAML = require("yaml");
 const Schema = require("./schema");
 
-function Config({env, configFile}) {
-  if (!fs.existsSync(configFile)) {
-    throw new Error(`THE RESTQA CONFIG FILE IS MISSING (${configFile})`);
+class Config {
```

Comment:
A lot of classes in this file. Maybe could we create a `src/config/internals.js` file?